Yubico.YubiKey/src/Yubico/YubiKey/Cryptography/AesUtilities.cs

```diff
@@ -152,14 +151,14 @@ public static byte[] AesCbcEncrypt(byte[] key, byte[] iv, ReadOnlySpan<byte> pla
         /// <param name="iv">16-byte initialization vector (IV)</param>
         /// <param name="ciphertext">Input blocks; must be a non-zero multiple of 16 bytes long</param>
         /// <returns>Plaintext of the same length as the ciphertext</returns>
-        public static byte[] AesCbcDecrypt(byte[] key, byte[] iv, ReadOnlySpan<byte> ciphertext)
+        public static ReadOnlyMemory<byte> AesCbcDecrypt(ReadOnlySpan<byte> key, ReadOnlySpan<byte> iv, ReadOnlySpan<byte> ciphertext)
         {
-            if (key is null)
+            if (key == null)
```

Comment:
In the previous method you use `IsEmpty` property of ReadOnlySpan - but here you nullcheck. What is the difference?

Reply:
https://stackoverflow.com/a/50811687

It's similar to some language's `===` operator. It takes things like type into account. For reference types in C#, that's important. `==` can potentially be overloaded, though this code base rarely, if ever, does this. I also think, given how C# treats null value types differently from null reference types, `is` provides some syntactic sugar to smooth out those differences and make null checking the same across the two.

In my opinion the `is null` is the more "modern" and generally "safer" (i.e. fewer surprises) way to perform true null checking.

Reply:
Yes, I prefer null as well. Over the past months many iterations have passed. This change was not intentional.

Reply:
Ah, right - you were asking about it compared to Empty. Yes, Dennis is 100% right. ReadOnlySpan/Memory you wanna go with an Empty check most of the times.